addons/binding/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/handler/NukiBridgeHandler.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.nuki.handler;
+
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.nuki.internal.dataexchange.BridgeInfoResponse;
+import org.openhab.binding.nuki.internal.dataexchange.NukiHttpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NukiBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Katter - Initial contribution
+ */
+public class NukiBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(NukiBridgeHandler.class);
+
+    private NukiHttpClient nukiHttpClient;
+
+    public NukiBridgeHandler(Bridge bridge) {
+        super(bridge);
+        logger.trace("Instantiating NukiBridgeHandler({})", bridge);
+    }
+
+    public NukiHttpClient getNukiHttpClient() {
+        return nukiHttpClient;
+    }
+
+    @Override
+    public void initialize() {
+        scheduler.execute(() -> initializeHandler());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("NukiBridgeHandler:handleCommand({}, {})", channelUID, command);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("NukiBridgeHandler:dispose()");
+        nukiHttpClient.stopClient();
+        // nukiHttpClient = null;
+    }
+
+    private void initializeHandler() {
+        logger.debug("NukiBridgeHandler:initializeHandler()");
+        nukiHttpClient = new NukiHttpClient(getConfig());
```

Comment:
You could consider (or just test) to move this method out of here and into `initialize`, so set-up the basis connection in the main thread and spin-up a new thread to check whether it's successful?

Don't forget then to also synchronize the block in the `initialize` method.